### ISSUE TITLE
Customizable Rollups: Rollups Fail When >1 Opportunity/Payment Relationship

### ISSUE THREAD

**Issue by jesslopez-sf** [SalesforceFoundation/Cumulus#3345]
When Payment Rollups exist and there is more than one Payment => Opportunity Relationship, all rollups fail to calculate. The error returned is:

`SObject row was retrieved via SOQL without querying the requested field: Opportunity.npe01__OppPayment__r Rollups State:`

**StackTrace:**
```
Stacktrace available at the time of saving the record - Class.CRLP_RollupProcessor_SVC.splitChildRecordsInMapByParentId: line 359, column 1
Class.CRLP_RollupQueueable.runRollupsForIds: line 148, column 1
Class.CRLP_RollupQueueable.execute: line 76, column 1
```

**Steps to Reproduce:**

1. Enable Customizable Rollups.

2. Create a Currency field on the Account called "Total Payment Amount".

3. Create a new Customizable Rollup:  (Target Object = Account, Target Field = Total Payment Amount, Operation = Sum, Time Frame = All Time, Rollup Type = Payment -> Account (Hard Credit), Filter Group = Pmts: Paid,  Amount Field = Payment: Payment Amount)

4. Create a Contact.

5. Create a Closed/Won Opportunity for that Contact/Household.  Amount = $100

6. Clear the Total Payment Amount field on the Household Account if it is filled in. Click the Recalculate Rollups button on the Household Account. The Total Payment Field = $100. [May need to refresh a few times in Lightning]

7. Clear out the Total Payment Amount field

8. Create a Lookup field on the Payment Object called Related Opportunity. The relationship should be to the Opportunity Object.

9. Go back to the Household Account and click Recalculate Rollups. The Error can be viewed in the NPSP Error Log.

**Comment by jesslopez-sf** [comment]
**lurch: add

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x8000000AYBMTAA5" target="_blank">W-025873</a>

**Comment by mrbelvedere** [comment]
Included in production release 3.135
